src/jdk.jlink/share/classes/jdk/tools/jlink/internal/ImageFileCreator.java

```diff
@@ -223,7 +223,6 @@ private IOException newIOException(RuntimeImageLinkException e) throws IOExcepti
             e.printStackTrace();
         }
         String message = switch (e.getReason()) {
-            case PATCH_MODULE -> helper.getMessage("err.runtime.link.patched.module", e.getFile());
```

Comment:
As `MODIFIED_FILE` is the only reason to throw RILE, the enum `Reason` is no longer needed.  This can be simplified.

Reply:
Yes, the updated patch has this removed.